### PR TITLE
Add support for output file naming (resolves #57)

### LINKED ISSUE #57: Add new argument to wrapper for output file basename
Add a new argument to the wrapper that will specify the basename to use when naming the output files. This is needed to support Dockstore and CWL where predictable output file names are required and to make the output names more meaningful to the bioinformaticians

### PR DESCRIPTION
**Pull request by wshands**
Added support for basename for output file naming @jvivian 

Fixed most comments in Thomas' pull request. The ones I did not:
  I believe the '-f' switch for docker tag has been deprecated (check with CJ) so I removed it; it also failed in my tests with the '-f' but worked without it
  I left the original code for getSampleName which split on a '.' to try an minimize side effects of code changes

### CODE REVIEW

**Comment by jvivian** [comment]
Thanks @wshands — I know you're on a time crunch, so contingent upon jenkins building successfully I'll merge this.

**Comment by wshands** [comment]
OK one thing is that with these changes I believe the wrapper will always remove the 'FAIL.' prefix, not sure if this is what we want...is it OK with you?

File: docker/wrapper.py [hunk 1]
```diff
@@ -56,8 +56,25 @@ def call_pipeline(mount, args):
         else:
             log.info('Flag "--no-clean" was used, therefore {} was not deleted.'.format(work_dir))
 
+    log.info("output dir is {} and files are:\n{}:".format(mount,'\n'.join(os.listdir(mount))))
+    fail_files = [x for x in os.listdir(mount) if x.startswith('FAIL.')]
+    log.info("fail files are:\n{}:".format('\n'.join(fail_files)))
+    for fail_file in fail_files:
+        new_file = fail_file.lstrip('FAIL.')
```
Comment (jvivian) [inline docker/wrapper.py]: Sorry, I misled you here. `lstrip` has some unintended side effects. Can you change to `new_file = fail_file[5:]` ?

Comment (wshands) [inline docker/wrapper.py]: OK

**Comment by jvivian** [comment]
> OK one thing is that with these changes I believe the wrapper will always remove the 'FAIL.' prefix, not sure if this is what we want...is it OK with you?

That's what you want — the only possible conflict would be if a user submits a sample prepended with *FAIL.*, in which case it would be stripped off.

File: docker/wrapper.py (hunk 1 above)
Comment (wshands) [inline docker/wrapper.py]: How about new_file = fail_file[len('FAIL.'):]
so no one misses it if they change 'FAIL' above the line?

Comment (jvivian) [inline docker/wrapper.py]: You'd need to assign "FAIL." to a variable if you wanted that effect.  i.e.

```
tag = 'FAIL.'
fail_files = [x for x in os.listdir(mount) if x.startswith(tag)]
new_file = fail_file[len(tag):]
```

**Comment by wshands** [comment]
OK I replaced strip with slice for new file name

**Comment by jvivian** [comment]
Thanks @tboser @wshands — I'll go ahead and merge, cherrypick this commit over to the release branch and push out a new release **3.1.2**.

**Pull request merged**